### DISCUSSION TITLE
redis cache with pagination dont work

### DISCUSSION THREAD

**Discussion by zatrudnitel** [livewire/livewire#5706] [1. Help] [answered]
Hello all! I use livewire with pagination. If I do not use the cache, then everything works. If I enable redis cache then pagination stops working. The ajax post request is made, but the data is not displayed. Do not tell me what could be the problem?

My livewire control:
```
<?php

namespace App\Http\Livewire;

use Illuminate\Http\Request;
use Illuminate\Support\Facades\DB;
use Illuminate\Support\Facades\Cache;
use Livewire\Component;
use Livewire\WithPagination;

class ItemList extends Component
{
    use WithPagination;
    protected $paginationTheme = 'bootstrap';

    public function render()
    {
        $page = request()->page;
        $blockID = array();

            $items = Cache::tags(['items'])->remember('items'.$page, 600, function () {
                return DB::table('items')
                    ->select('items.id', 'items.name', 'items.shortName')
                    ->paginate(20);
            });

            foreach ($items as $key){
                array_push($blockID, $key->id);
            }

            $prices = Cache::tags(['prices'])->remember('prices'.$page, 600, function () use ($blockID) {
                return DB::table('itemprices')
                    ->select('itemprices.item', 'traders.name as tname', 'traders.imageLink', 'itemprices.act')
                    ->join('currency','currency.id','=','itemprices.currency')
                    ->join('traders','traders.id','=','itemprices.trader')-whereIn('itemprices.item', $blockID)->get();
            });
      

        return view('livewire.item-list', ['items' => $items, 'prices' => $prices]);
    }
}
```

my .env
```
BROADCAST_DRIVER=log
CACHE_DRIVER=redis
FILESYSTEM_DISK=local
QUEUE_CONNECTION=sync
SESSION_DRIVER=file
SESSION_LIFETIME=180

REDIS_HOST=127.0.0.1
REDIS_PASSWORD="pass"
REDIS_PORT=6379
```
![image](https://github.com/livewire/livewire/assets/29029547/16c9a75b-f042-4a1f-99e6-0108971002d2)
in debug bar no sql query...

**Comment by chatisk** [comment]
Have you found any solution? Did you solve it somehow? @DifficultPhil

**Comment by lrljoe** [comment, marked as answer]
If you cache a paginated item, then you're always going to get the same set of results, as it won't actually understand pagination, unless you key correctly, and interface with the pagination methods.

Otherwise, don't cache paginated, it's not a Livewire issue, look at your cached data for clarity.

If you want to cache, then you need to use keys to tell it which page to return from the cache, and ensure you're telling it which page to return.

Keep in mind this stops users from changing perPage etc, and is not recommended.

```
$items = Cache::tags(['items'])->remember('items'.$page, 600, function () {
                return DB::table('items')
                    ->select('items.id', 'items.name', 'items.shortName')
                    ->paginate(20);
            });
```

Here, you are telling it to return the first 20 results, regardless of the page number.  You need to pass in the correct page into the query.
